utils/types/models.ts

```diff
@@ -31,13 +31,31 @@ export interface Category {
   name: string
 }
 
+export type AttributePossibleValues = 'text' | 'number' | string[]
+
 export interface Attribute {
   _id?: string
   name: string
-  possibleValues: 'text' | 'number' | string[]
+  possibleValues: AttributePossibleValues
   color: string
 }
 
+export interface OptionsAttribute extends Attribute {
+  possibleValues: string[]
+}
+
+export interface TextAttribute extends Attribute {
+  possibleValues: 'text'
+}
+
+export interface NumberAttribute extends Attribute {
+  possibleValues: 'number'
+}
+
+export interface AttributeRequest {
```

Comment:
Is this really necessary? we have a mongo driver func to get an entity by id and another to find by a more complex filter. Just don't want to get code that's longer/less readable than needs to be